packages/components/src/components/chat-search/Context.vue

```diff
@@ -15,7 +15,7 @@
     <template v-else>
       <div class="context__body">
         <h2>Context</h2>
-        <template v-if="pinnedItems && pinnedItems.length">
+        <template v-if="hasPinnedItems">
           <h3>Pinned Items</h3>
           <div class="context__body__table">
             <component
```

Comment:
Somehow the context is being written to an attribute as a literal, but I don't see that change here.
![image](https://github.com/user-attachments/assets/4cdd8397-d0fd-4373-98a5-ec0fbf405991)

Reply:
The code that renders pinned items assumes that each item has a `content` property: https://github.com/getappmap/appmap-js/blob/pin-file_20240825/packages/components/src/components/chat/PinEvent.ts#L10 . When a file gets pinned, the `PinFile.content` for it gets set to the content of the file.

The other changes I made in this file insure that the content for a pinned file doesn't get revealed when you click on it.